apis/kueue/v1beta1/clusterqueue_conversion.go

```diff
@@ -38,12 +38,56 @@ func (dst *ClusterQueue) ConvertFrom(srcRaw conversion.Hub) error {
 
 func Convert_v1beta1_ClusterQueueSpec_To_v1beta2_ClusterQueueSpec(in *ClusterQueueSpec, out *v1beta2.ClusterQueueSpec, s conversionapi.Scope) error {
 	out.CohortName = v1beta2.CohortReference(in.Cohort)
-	return autoConvert_v1beta1_ClusterQueueSpec_To_v1beta2_ClusterQueueSpec(in, out, s)
+	if err := autoConvert_v1beta1_ClusterQueueSpec_To_v1beta2_ClusterQueueSpec(in, out, s); err != nil {
+		return err
+	}
+
+	// Convert AdmissionChecks to AdmissionChecksStrategy
+	// If v1beta1 has AdmissionChecks but v1beta2 doesn't have AdmissionChecksStrategy,
+	// convert the old field to the new structure
+	if len(in.AdmissionChecks) > 0 && out.AdmissionChecksStrategy == nil {
+		out.AdmissionChecksStrategy = &v1beta2.AdmissionChecksStrategy{
+			AdmissionChecks: make([]v1beta2.AdmissionCheckStrategyRule, len(in.AdmissionChecks)),
+		}
+		for i, checkRef := range in.AdmissionChecks {
+			out.AdmissionChecksStrategy.AdmissionChecks[i] = v1beta2.AdmissionCheckStrategyRule{
+				Name:      v1beta2.AdmissionCheckReference(checkRef),
+				OnFlavors: []v1beta2.ResourceFlavorReference{}, // Empty means all flavors
+			}
+		}
+	}
+
+	return nil
 }
 
 func Convert_v1beta2_ClusterQueueSpec_To_v1beta1_ClusterQueueSpec(in *v1beta2.ClusterQueueSpec, out *ClusterQueueSpec, s conversionapi.Scope) error {
 	out.Cohort = CohortReference(in.CohortName)
-	return autoConvert_v1beta2_ClusterQueueSpec_To_v1beta1_ClusterQueueSpec(in, out, s)
+	if err := autoConvert_v1beta2_ClusterQueueSpec_To_v1beta1_ClusterQueueSpec(in, out, s); err != nil {
```

Comment:
I don't think this conversion requires any changes. v1beta1 supports both AdmissionChecks and AdmissionCheckStrategy.